kolibri/plugins/learn/assets/src/state/actions.js

```diff
@@ -447,10 +447,17 @@ function showSearch(store, channelId, searchTerm) {
 }
 
 function showExamList(store, channelId) {
-  store.dispatch('CORE_SET_PAGE_LOADING', true);
+  const userIsLoggedIn = coreGetters.isUserLoggedIn(store.state);
   store.dispatch('SET_PAGE_NAME', PageNames.EXAM_LIST);
+  store.dispatch('CORE_SET_PAGE_LOADING', true);
 
-  coreActions.setChannelInfo(store, channelId).then(
+  // if user is not logged in, this action is a noop
+  if (!userIsLoggedIn) {
+    store.dispatch('CORE_SET_PAGE_LOADING', false);
```

Comment:
This is in a category of route/action handling that would be great to approach more systematically than we currently are. At a high level, I think we have a need for middleware wrappers around routing actions to handle cases like:

* 'not found'-type issues (e.g. content URLs, reports for invalid IDs, etc)
* authorization issues (e.g. loading a manage page as a regular user)
* general uncaught errors (current fallback on `SET_ERROR`

A lot of these are kind of the client-side version of what you'd expect to see from a properly configured web server (404s, 403s, 500s...) and we might be able to think about handling them similarly (e.g. adding "decorators" to various actions specifying what kind of user it's applicable for)

In general, I'd like to avoid redirecting for any of these because it dissociates the URL that caused the issue with the result. It means if someone wants to e.g. refresh the page to try again, it won't work. Instead, we should move towards incorporating common scenarios into `core-base` and handling the scenarios consistently across routing actions.